distributed/scheduler.py

```diff
@@ -3334,6 +3340,20 @@ def get_profile_metadata(self, comm=None, workers=None, merge_workers=True,
 
         raise gen.Return({'counts': counts, 'keys': keys})
 
+    def get_logs(self, comm=None, n=None):
+        if n is None:
+            L = list(deque_handler.deque)
+        else:
+            L = deque_handler.deque
+            L = [L[-i] for i in range(min(n, len(L)))]
```

Comment:
Not sure, but `list(deque_handler.deque)[-n:]` looks simpler (and potentially faster) to me.

Reply:
It's only faster above 1000 elements,.  I suspect that most entries for n will be fairly small (otherwise people will just use `n=None`)

```python
In [1]: from collections import deque

In [2]: d = deque(range(10000))

In [3]: %timeit [d[-i] for i in range(100)]
100000 loops, best of 3: 8.78 µs per loop

In [4]: %timeit [d[-i] for i in range(1000)]
The slowest run took 4.67 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 88.7 µs per loop

In [5]: %timeit [d[-i] for i in range(10000)]
1000 loops, best of 3: 1.52 ms per loop

In [6]: %%timeit
   ...: L = list(d)
   ...: [L[-i] for i in range(100)]
   ...: 
The slowest run took 4.87 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 54.9 µs per loop

In [7]: %%timeit
   ...: L = list(d)
   ...: [L[-i] for i in range(1000)]
   ...: 
10000 loops, best of 3: 128 µs per loop

In [8]: %%timeit
   ...: L = list(d)
   ...: [L[-i] for i in range(10000)]
   ...: 
The slowest run took 4.30 times longer than the fastest. This could mean that an intermediate result is being cached.
1000 loops, best of 3: 684 µs per loop
```